### PR TITLE
Replaced deprecated org.apache.commons.lang library

### PR DESCRIPTION
**Pull request by namigupt**
1. Replaced deprecated 'org.apache.commons.lang' library with 'org.apache.commons.text'
2. Added Serial Version UID.

### CODE REVIEW

**Comment by namigupt** [comment]
Hi @rombert ,

Please review the PR.

Regards,
Namit

File: pom.xml [hunk 1]
```diff
@@ -16,308 +16,309 @@
   ~ and limitations under the License.
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~-->
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
-    <modelVersion>4.0.0</modelVersion>
```
Comment (rombert) [inline pom.xml]: Please revert the whitespace-only changes done to the pom file. It makes it hard to review  what has actually changed.

File: src/main/java/javax/xml/parsers/FactoryFinder.java [hunk 2]
```diff
@@ -159,7 +159,9 @@ static Object find(String factoryId, String fallbackClassName)
     }
 
     static class ConfigurationError extends Error {
-        private Exception exception;
+        //Serial Version UID
+		private static final long serialVersionUID = 159720375000853069L;
```
Comment (rombert) [inline src/main/java/javax/xml/parsers/FactoryFinder.java]: Why is this needed?

File: src/main/java/org/apache/sling/xss/impl/webconsole/XSSProtectionAPIWebConsolePlugin.java [hunk 3]
```diff
@@ -58,7 +58,9 @@
 )
 public class XSSProtectionAPIWebConsolePlugin extends HttpServlet {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(XSSProtectionAPIWebConsolePlugin.class);
+    //Serial Version UID
+	private static final long serialVersionUID = -8510989967759120654L;
```
Comment (rombert) [inline src/main/java/org/apache/sling/xss/impl/webconsole/XSSProtectionAPIWebConsolePlugin.java]: Why is this needed?

**Review by rombert — CHANGES_REQUESTED** [review]
Thanks for the PR @namigupt ! I made a couple of comments, it would also be great if you could file a Jira issue under https://issues.apache.org/jira/browse/SLING to track this change.

File: pom.xml [hunk 4]
```diff
@@ -16,308 +16,309 @@
...
+			<version>1.0-alpha-1</version>
+			<scope>provided</scope>
+		</dependency>
+		<dependency>
+			<groupId>commons-io</groupId>
+			<artifactId>commons-io</artifactId>
+			<version>2.6</version>
+			<scope>provided</scope>
+		</dependency>
+		<!-- https://mvnrepository.com/artifact/org.apache.commons/commons-text -->
```
Comment (klcodanr) [inline pom.xml]: Unnecessary comment, can be removed

**Comment by namigupt** [comment]
@rombert  @klcodanr 

Thanks for the review. I have incorporated the requisite changes.
Also, a JIRA has been raised to track these changes - [SLING-11201](https://issues.apache.org/jira/browse/SLING-11201)

File: src/main/java/javax/xml/parsers/FactoryFinder.java [hunk 5]
```diff
@@ -159,7 +159,8 @@ static Object find(String factoryId, String fallbackClassName)
     }
 
     static class ConfigurationError extends Error {
-        private Exception exception;
+       
+		private Exception exception;
```
Comment (rombert) [inline src/main/java/javax/xml/parsers/FactoryFinder.java]: nit: please remove this whitespace-only change

File: src/main/java/org/apache/sling/xss/impl/webconsole/XSSProtectionAPIWebConsolePlugin.java [hunk 6]
```diff
@@ -58,7 +58,7 @@
 )
 public class XSSProtectionAPIWebConsolePlugin extends HttpServlet {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(XSSProtectionAPIWebConsolePlugin.class);
+	private static final Logger LOGGER = LoggerFactory.getLogger(XSSProtectionAPIWebConsolePlugin.class);
```
Comment (rombert) [inline src/main/java/org/apache/sling/xss/impl/webconsole/XSSProtectionAPIWebConsolePlugin.java]: nit: please remove this whitespace-only change

File: pom.xml [hunk 7]
```diff
@@ -273,11 +273,11 @@
             <scope>provided</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.commons</groupId>
-            <artifactId>commons-lang3</artifactId>
-            <version>3.6</version>
-            <scope>provided</scope>
-        </dependency>
+			<groupId>org.apache.commons</groupId>
```
Comment (rombert) [inline pom.xml]: nit: please make sure they indentation does not change compared to the other entries in the pom.xml file.

**Review by rombert — CHANGES_REQUESTED** [review]
Thanks for the update @namigupt . We still have a few whitespace issues in the pom and in the Java files. The GitHub UI shows then in the PR. I also find it useful to run locally before submitting something like

```
$ git diff origin/master...
```

This allows me to catch whitespace issues early on.

**Comment by namigupt** [comment]
@rombert - formatting issues have been fixed now.

**Comment by rombert** [comment]
Thanks for the rapid turnaround @namigupt ! I merged your changes squashed into a single commit to make tracking history simpler.
